common/agentinstall/install.go

```diff
@@ -24,7 +26,11 @@ func Install(runner *command.Runner, commonNamer namer.Namer, params *Params, os
 	if params.agentConfig != "" {
 		fileManager := command.NewFileManager(runner)
 		remotePath := os.GetConfigPath()
-		lastCommand, err = fileManager.CopyInlineFile("agent-config", pulumi.String(params.agentConfig), remotePath, true, pulumi.DependsOn([]pulumi.Resource{lastCommand}))
+		agentConfig := env.AgentAPIKey().ApplyT(func(apiKey string) pulumi.String {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Does `ApplyT` block and wait for the resolution of the string (yes I could read the docs)

Reply:
Read the doc, it's more like a `.then`, returning a new ~Promise~ Output